.github/workflows/validation.yaml

```diff
@@ -0,0 +1,39 @@
+name: Kubernetes Validation
+
+on:
+  pull_request:
+    branches:
+      - main
```

Comment:
I would prefer to run it on all PRs and on all commits on the main branch instead of only PRs targeting the main branch and probably all tags.